### PR TITLE
Improve handling of errors while fetching build history

### PR DESCRIPTION
**Pull request by arjunsalyan**
https://github.com/macports/macports-webapp/issues/273

This skips the current build if 404 response code is recieved from buildbot, otherwise it stops the process for the builder so that a retry attempt is made in the next run of the cronjob.

### CODE REVIEW

**Review by mojca — APPROVED** [review]
This looks ok to me.

It maybe wouldn't hurt to have a simple test, but it's already good as it is.

File: app/ports/models/buildhistory.py [hunk 1]
```diff
@@ -99,7 +102,9 @@ def load_database(data):
 
         for buildername in builders:
             # fetch the last build first in order to figure out its number
-            last_build_data = get_data_from_url(get_url_json(buildername, -1))
+            last_build_data, resp_code = get_data_from_url(get_url_json(buildername, -1))
```
Comment (neverpanic) [inline app/ports/models/buildhistory.py:105]:
```suggestion
            last_build_data, _ = get_data_from_url(get_url_json(buildername, -1))
```
It's customary to use `_` for unused variables in Python.

**Review by neverpanic — APPROVED** [review]
A nitpick, lg2m otherwise.

File: app/ports/models/buildhistory.py (hunk 1 above)
Comment (ryandesign) [reply app/ports/models/buildhistory.py:105]: I tried to commit this suggestion but got "Invalid email for web commit. Refresh and try again." If anyone is able to commit this suggestion please do, then let's squash and merge the PR.

Comment (mojca) [reply app/ports/models/buildhistory.py:105]: I'm sorry, I misread the contents and for some reason thought that this has already been fixed (while it was still just a suggestion). Anyway, I merged this now, we can still do this cosmetic change in a separate commit, hoping that the site will be fixed now.

**Pull request merged**